src/Symfony/Component/Security/Core/Authentication/Token/RememberMeToken.php

```diff
@@ -49,25 +42,20 @@ public function getFirewallName(): string
         return $this->firewallName;
     }
 
-    /**
-     * @deprecated since Symfony 7.2
-     */
-    public function getSecret(): string
-    {
-        trigger_deprecation('symfony/security-core', '7.2', 'The "%s()" method is deprecated.', __METHOD__);
-
-        return $this->secret ??= base64_encode(random_bytes(8));
-    }
-
     public function __serialize(): array
     {
-        // $this->firewallName should be kept at index 1 for compatibility with payloads generated before Symfony 8
-        return [$this->secret, $this->firewallName, parent::__serialize()];
+        // Newly serialized data removes the secret deprecated since Symfony 7.2 and removed in Symfony 8.0
+        return [$this->firewallName, parent::__serialize()];
     }
 
     public function __unserialize(array $data): void
     {
-        [$this->secret, $this->firewallName, $parentData] = $data;
+        // BC Layer with payloads generated before Symfony 8.0
```

Comment:
```suggestion
        // BC Layer with payloads generated before Symfony 8.0, to be removed in Symfony 9.0
```

Is it ok to keep such a BC Layer here? @OskarStark @stof

Reply:
Works for me

Reply:
it is even a very good idea to have it IMO.